src/MerchantCenter/MerchantStatuses.php

```diff
@@ -178,7 +199,14 @@ public function get_issues( string $type = null, int $per_page = 0, int $page =
 	 * @since 1.1.0
 	 */
 	public function clear_cache(): void {
-		$this->container->get( TransientsInterface::class )->delete( TransientsInterface::MC_STATUSES );
+		$update_all_products_job = $this->container->get( UpdateAllProducts::class );
+		$delete_all_products_job = $this->container->get( DeleteAllProducts::class );
+
+		// Clear the cache if we are not in the middle of updating/deleting all products. Otherwise, we might update the product stats for each individual batch.
+		// See: ClearProductStatsCache::register
+		if ( $update_all_products_job->can_schedule( null ) && $delete_all_products_job->can_schedule( null ) ) {
```

Comment:
Why `can_schedule` and not `is_running`?

Reply:
Because `is_running` is a protected method, I understand that maybe it would be a bit easier to read the code if we use a method name similar to `is_running`. Although I think the comment that I left explains what we are doing.

Reply:
More than the naming I was more thinking about the logic in can_schedule. That also checks some conditions. If MC connected for example. But if that doesn't create a problem I think we are good with can_schedule